ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,406 @@
+  const t = useI18nContext();
+
+  const [hasScrolledMessage, setHasScrolledMessage] = useState(false);
+  const [showContractDetails, setShowContractDetails] = useState(false);
+  const [messageRootRef, setMessageRootRef] = useState(null);
+  const {
+    type,
+    msgParams: { from, data, origin, version },
+  } = txData;
+
+  const {
+    chainId,
+    type: providerName,
+    rpcPrefs,
+    nickname: providerNickName,
+  } = useSelector(getProviderConfig);
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const hardwareWalletRequiresConnection = useSelector((state) =>
+    doesAddressRequireLedgerHidConnection(state, from),
+  );
+  const isLedgerWallet = useSelector((state) => isAddressLedger(state, from));
+  const unconfirmedMessagesList = useSelector(unconfirmedMessagesHashSelector);
+  const unapprovedMessagesCount = useSelector(getTotalUnapprovedMessagesCount);
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const conversionRateSelected = useSelector(conversionRateSelector);
+  const conversionRate = useNativeCurrencyAsPrimaryCurrency
+    ? null
+    : conversionRateSelected;
+  const subjectMetadata = useSelector(getSubjectMetadata);
+  // not forwarded to component
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+  const { address, balance, name } =
+    getAccountByAddress(allAccounts, from) || {};
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  // Used to show a warning if the signing account is not the selected account
+  // Largely relevant for contract wallet custodians
+  const selectedAccount = useSelector(getSelectedAccount);
+  ///: END:ONLY_INCLUDE_IN
+
+  let cancel;
+  let sign;
+
+  if (type === MESSAGE_TYPE.PERSONAL_SIGN) {
+    cancel = cancelPersonalMessage;
+    sign = signPersonalMessage;
+  } else if (type === MESSAGE_TYPE.ETH_SIGN_TYPED_DATA) {
+    cancel = cancelTypedMessage;
+    sign = signTypedMessage;
+  } else if (type === MESSAGE_TYPE.ETH_SIGN) {
+    cancel = cancelMessage;
+    sign = signMessage;
+  }
+
+  const memoizedParseMessage = memoize(() => {
+    const { message, domain = {}, primaryType, types } = JSON.parse(data);
+    const sanitizedMessage = sanitizeMessage(message, primaryType, types);
+    return { sanitizedMessage, domain, primaryType };
+  });
+
+  const getNetworkName = (providerType, nickName) => {
+    switch (providerType) {
+      case NETWORK_TYPES.MAINNET:
+        return t('mainnet');
+      case NETWORK_TYPES.GOERLI:
+        return t('goerli');
+      case NETWORK_TYPES.SEPOLIA:
+        return t('sepolia');
+      case NETWORK_TYPES.LINEA_TESTNET:
+        return t('lineatestnet');
+      case NETWORK_TYPES.LOCALHOST:
+        return t('localhost');
+      default:
+        return nickName || t('unknownNetwork');
+    }
+  };
```

Comment:
Above can be just util fn.

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/19160